src/source/geojson_worker_source.js

```diff
@@ -293,4 +296,70 @@ class GeoJSONWorkerSource extends VectorTileWorkerSource {
     }
 }
 
+function getSuperclusterOptions({superclusterOptions, clusterProperties}) {
+    if (!clusterProperties || !superclusterOptions) return superclusterOptions;
+
+    const initialValues = {};
+    const mapExpressions = {};
+    const reduceExpressions = {};
+    const globals = {accumulated: null, zoom: 0};
+    const feature = {properties: null};
+    const propertyNames = Object.keys(clusterProperties);
+
+    for (const key of propertyNames) {
+        const [operator, initialExpression, mapExpression] = clusterProperties[key];
+
+        const parsed = createExpression(clusterProperties[key]);
+        if (parsed.result === 'error') {
+            const message = parsed.value.map(e => e.message).join('; ');
+            throw new Error(`Error parsing expression for cluster property "${key}": ${message}`);
+
+        } else if (!isGlobalPropertyConstant(parsed.value.expression, ['zoom'])) {
+            throw new Error(`Error parsing expression for cluster property "${key}": zoom expressions not supported.`);
+        }
+
+        const initialExpressionParsed = createExpression(initialExpression);
+        const mapExpressionParsed = createExpression(mapExpression);
+        const reduceExpressionParsed = createExpression([operator, ['accumulated'], ['get', key]]);
+
+        if (initialExpressionParsed.result === 'success') {
+            if (!isFeatureConstant(initialExpressionParsed.value.expression))
+                throw new Error(`Error parsing expression for cluster property "${key}": can't use feature properties in initial expression.`);
+
+            initialValues[key] = initialExpressionParsed.value.evaluate(globals);
+        }
+
+        if (mapExpressionParsed.result === 'success')
+            mapExpressions[key] = mapExpressionParsed.value;
+
+        if (reduceExpressionParsed.result === 'success')
+            reduceExpressions[key] = reduceExpressionParsed.value;
+    }
+
+    superclusterOptions.initial = () => {
+        const properties = {};
+        for (const key of propertyNames) {
+            properties[key] = initialValues[key];
+        }
+        return properties;
+    };
+    superclusterOptions.map = (pointProperties) => {
+        feature.properties = pointProperties;
+        const properties = {};
+        for (const key of propertyNames) {
+            properties[key] = mapExpressions[key].evaluate(globals, feature);
+        }
+        return properties;
+    };
+    superclusterOptions.reduce = (accumulated, clusterProperties) => {
+        feature.properties = clusterProperties;
+        for (const key of propertyNames) {
+            globals.accumulated = accumulated[key];
```

Comment:
Currently not, but I think this could be added (check if `operator` is string and if it's not, consider it as a reduce expression). Would add quite a bit of complexity to the docs, but add some flexibility as well. Not sure how to go here.

Reply:
Ah ok, I see - the reduce expression has to be something that fits into `[operator, ['accumulated'], ['get', key]]`. So that's why `max` and `any` work.

👍